src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Config/EventHubOptions.cs

```diff
@@ -281,6 +283,22 @@ internal EventProcessorHost GetEventProcessorHost(IConfiguration config, string
             throw new InvalidOperationException("No event hub receiver named " + eventHubName);
         }
 
+        // Lookup EventHub endpoint given the name provided in the [EventHubTrigger] attribute. 
+        internal string GetEventHubEndpoint(string eventHubName)
+        {
+            if (_receiverCreds.TryGetValue(eventHubName, out var creds))
+            {
+                return creds.EventHubEndpoint;
+            }
+
+            if (_explicitlyProvidedHosts.TryGetValue(eventHubName, out var host))
+            {
+                return host.EndpointAddress.OriginalString;
+            }
+
+            return null;
+        }
```

Comment:
This will be the same every time it's called, right? If so we should cache it so we don't have to go through this every time.

Reply:
it's cached in the `_receiverCreds` (or `_explicitlyProvidedHosts` dictionaries), so this is lookup in the dictionary that most likely contains one element.

As far as I understand there could be multiple event hubs and EventProcessor only knows EventHub name when it starts processing the message here:

https://github.com/Azure/azure-functions-eventhubs-extension/blob/e606dfd79683b03b9cd96f5ff49fe4ecec4ef607/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs#L169

So in theory, `EventProcessor` can be reused between different EventHubs and I'm not sure if it's ok to change it. If it's ok I can pass endpoint to the `EventProcessor` and cache as instance variable. 

I assumed lookup in the dictionary is a better alternative, happy to change if you think otherwise.

Reply:
Ahh, yes, makes sense!